prefq_documentation.md

```diff
@@ -0,0 +1,110 @@
+
+```python
+if __name__ == "__main__":
+    before_first_request()
+    app.run(host="localhost", port=5000, debug=True)
+```
+
+### Server
+Start your own local server:
+1. Clone the repository: `git clone https://github.com/timokau/prefq.git`
+2. Install poetry: ```bash curl -sSL https://install.python-poetry.org | python3 - ```
+3. Add poetry to PATH and make sure this persists across reboots. 
+      ```bash
+    export PATH="$HOME/.local/bin:$PATH"
+    echo 'export PATH="$HOME/bin:$PATH"' > ~/.bashrc```
+4. Start the server: `scripts/run_server.sh`
+
+### Query Client
+Send queries to your local server:
+1. Open a new terminal
+2. *(a)* Send static queries: `scripts/run_static_queries.sh`
+    *(b)* Send dynamic queries: `scripts/run_imitation_example.sh`
+
+
+Option (a):
+- shows the required http-request structure for sending queries & receiving feedback
+- does not learn new behavior
+
+Option (b):
+- integrates our PrefQ-Server into [imitation](https://pypi.org/project/imitation/), a library based on [gymnasium](https://gymnasium.farama.org/index.html)
+- capable of learning new behavior
+
+### Feedback Client
+Evaluate queries:
+1. Visit the web interface: `http://localhost:5000` (*using your favorite web browser*)
+2. Select your preferred video: use left/right arrow keys or click the corresponding button
+
+Using such a development server holds several eminent advantages
+1. Changes within your project folder will be recognized. As soon as you save changes, the server will reboot automatically and you can instantly test and verify new behavior.
+2. Valuable debug information about incoming http-requests will be provided.
+3. You can develop your own server without being connected to the web.
+
+
+## Production Stage Server (remote)
+
+For remote feedback collection it is also possible to run a development stage server remotely. To do so you can replace `"localhost"` with the default route `"0.0.0.0"`. This will allow incoming http requests from any IP address.
+
+However, given that the Developement Server provided by Flask is "not designed to be particularly efficient, stable, or secure" [[1]](https://flask.palletsprojects.com/en/2.3.x/tutorial/deploy/) it is recommended to use a WSGI server instead. Additionally, in the Internet Connection Protocol (ICP), Port 80 is commonly used as the default network port for sending and receiving unencrypted web pages. [[2]](https://www.techopedia.com/definition/15709/port-80) This can be achieved by:
+
+1. installing waitress `pip install waitress`
+2. replacing the last lines of `server.py` as follows: 
+```python
+if __name__ == "__main__":
+    before_first_request()
+    from waitress import serve
+    serve(app, host='0.0.0.0', port=80)
```

Comment:
This is now out of sync with the defaults in the server file. Since you said that switching to waitress by default is not a good idea, I suggest making the backend configurable as well as the port and the host (#54).